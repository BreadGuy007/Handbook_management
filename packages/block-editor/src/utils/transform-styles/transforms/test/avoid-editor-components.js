/**
 * Internal dependencies
 */
import traverse from '../../traverse';
import avoidEditorComponents from '../avoid-editor-components';

describe( 'CSS selector wrap', () => {
	it( 'should add editor classes to buttons', () => {
		const callback = avoidEditorComponents();
		const input = `
		button {
			background-color: #ff0000;
		}`;
		const output = traverse( input, callback );

		expect( output ).toMatchSnapshot();
	} );

	it( 'should add editor classes to inputs', () => {
		const callback = avoidEditorComponents();
		const input = `
		input {
			border-color: #ff0000;
		}`;
		const output = traverse( input, callback );

		expect( output ).toMatchSnapshot();
	} );
} );
