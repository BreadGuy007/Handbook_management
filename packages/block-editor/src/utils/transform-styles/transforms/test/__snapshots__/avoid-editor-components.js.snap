// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSS selector wrap should add editor classes to buttons 1`] = `
"button:not(.components-button):not([id^=mceu_]) {
background-color: #ff0000;
}"
`;

exports[`CSS selector wrap should add editor classes to inputs 1`] = `
"input:not(.components-text-control__input):not(.components-placeholder__input):not(.components-form-token-field__input) {
border-color: #ff0000;
}"
`;
